base/binaryplatforms.jl

```diff
@@ -894,7 +894,7 @@ function detect_cxxstring_abi()
     end
 
     function open_libllvm(f::Function)
-        for lib_name in ("libLLVM", "LLVM", "libLLVMSupport")
+        for lib_name in ("libLLVM", "LLVM", "libLLVMSupport", "libLLVM-13jl")
```

Comment:
Looking through the rest, if we know it's generally going to be called `libLLVM-13jl`, I think we should try to dlopen that one first, and only fall back to `libLLVM` afterward.